edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/modules/RoutingModule.cs

```diff
@@ -547,5 +550,43 @@ protected override void Load(ContainerBuilder builder)
 
             base.Load(builder);
         }
+
+        static async Task<IEntityStore<TK, TV>> GetUpdatableEncryptedEntityStoreIfSupported<TK, TV>(IComponentContext context, string entityName)
+        {
+            var storeProvider = context.Resolve<IStoreProvider>();
+            Option<IEncryptionProvider> encryptionProvider = await context.Resolve<Task<Option<IEncryptionProvider>>>();
+            IEntityStore<TK, TV> entityStore = encryptionProvider.Map(
+                    e =>
+                    {
+                        IEntityStore<string, string> underlyingEntityStore = storeProvider.GetEntityStore<string, string>($"underlying{entityName}");
+                        IKeyValueStore<string, string> es = new UpdatableEncryptedStore<string, string>(underlyingEntityStore, e);
+                        ITypeMapper<TK, string> keyMapper = new JsonMapper<TK>();
+                        ITypeMapper<TV, string> valueMapper = new JsonMapper<TV>();
+                        IKeyValueStore<TK, TV> dbStoreMapper = new KeyValueStoreMapper<TK, string, TV, string>(es, keyMapper, valueMapper);
+                        IEntityStore<TK, TV> tes = new EntityStore<TK, TV>(dbStoreMapper, entityName);
+                        return tes;
+                    })
+                .GetOrElse(
+                    () => storeProvider.GetEntityStore<TK, TV>(entityName));
+
+            return entityStore;
+        }
+
+        async Task<IEntityStore<string, TwinStoreEntity>> GetTwinStore(IComponentContext context)
+        {
+            string entityName = "EdgeTwin";
+            IEntityStore<string, TwinStoreEntity> twinStore;
+            if (this.encryptTwinStore)
+            {
+                twinStore = await GetUpdatableEncryptedEntityStoreIfSupported<string, TwinStoreEntity>(context, entityName);
+            }
+            else
+            {
+                var storeProvider = context.Resolve<IStoreProvider>();
+                twinStore = storeProvider.GetEntityStore<string, TwinStoreEntity>(entityName);
```

Comment:
Nit: This else block and the `GetOrElse` block in `GetUpdatableEncryptedEntityStoreIfSupported` do the same thing. This path is taken when encryption was not requested, the other is taken when there's no encryption provider. Why not get the encrypted store if encryption is requested AND there's a provider, otherwise get the unencrypted store?

Reply:
Yes, I could.. I didn't to keep GetUpdatableEncryptedEntityStoreIfSupported generic, and because the common code was very simple. 
I can update it..

Reply:
Your call. I'm planning to hit "approve" once you've committed the comment about "updateable"...

Reply:
So I did this change, but it becomes a little ugly because encryptionProvider is an option. So to do the "is enabled and is supported", I have to do an expect inside the && and throw, and all. So I will keep it as it is. Especially since the else part is just storeProvider.GetEntityStore.. 
If you feel otherwise, lmk and I can refactor.

Reply:
No it's fine, keep it as-is.

Reply:
Okay, I think I found a good way to do this. :)